docs/upgrading.md

```diff
@@ -0,0 +1,111 @@
+# Upgrading Clear Containers installations
+
+This document discusses the available options for upgrading a system running Intel® Clear Containers.
+
+## Terminology
+
+There are currently two implementations of Clear Containers:
+
+- Clear Containers 2.x (or just 2.x) refers to the previous implementation
+  of Clear Containers which uses a runtime called `cc-oci-runtime` and
+  whose code is available at: https://github.com/01org/cc-oci-runtime
+
+- Clear Container 3.x (or just 3.x) refers to the current implementation
+  which uses a runtime called `cc-runtime` and whose code is available
+  at: https://github.com/clearcontainers/runtime
+
+## Maintenance warning
+
+The 2.x codebase is no longer being developed. New releases will *only* be
+considered for significant bugs.
+
+The main development focus is now on the 3.x branch. As such, all 2.x
+users are encouraged to switch to the 3.x implementation.
+
+## Introduction
+
+The two Clear Containers implementations share certain architectural
+elements. Namely, both use a shim process called `cc-shim` and a proxy
+process called `cc-proxy`. However, the 2.x versions of these
+elements *are incompatible* with their 3.x counterparts.
+
+## Supported operations and configurations
+
+### Systems using pre-built packages
+
+For operating systems that pre-built packages are provided for (such as
+Fedora* and Ubuntu*), upgrading is possible with some caveats:
+
+- Configuration files are not migrated (see [Configuration files](#configuration-files)).
+- It is necessary to perform a few cleanup steps as outlined in the
+  subsections below.
+
+Once the cleanup steps have been performed, simply follow the 3.x
+installation guides and any existing packaged versions of the 2.x
+components will be removed and the new 3.x components installed in their
+place.
+
+#### Upgrading an Ubuntu system
+
+Assuming the system was installed by following the [CC 2.x Ubuntu installation guide](https://github.com/01org/cc-oci-runtime/blob/master/documentation/Installing-Clear-Containers-on-Ubuntu.md):
+
+1. Perform the following cleanup steps:
+   ```bash
+   $ sudo rm -f /etc/systemd/system/docker.service.d/clr-containers.conf
+   $ sudo apt-get purge -y docker-engine
```

Comment:
@jodh-intel any reason you do not also reccomend apt-get remove and dnf remove of the packages that got isntalled

Reply:
We don't need to do that as the packaging changes will auto-remove the old CC 2.x packages when the user installs the new ones with `apt-get`/`dnf`. However, I found in testing that docker gets upset about being upgraded hence this entry.